crates/trie/sparse/src/state.rs

```diff
@@ -843,10 +845,10 @@ struct DecodedProofNodes {
     new_nodes: usize,
 }
 
-/// Decodes the proof nodes returning additional information about the number of total, skipped, and
-/// new nodes.
-fn decode_proof_nodes(
-    proof_nodes: ProofNodes,
+/// Does not decode the proof nodes, but instead just skips the nodes that are already revealed and
+/// returns additional information about the number of total, skipped, and new nodes.
+fn skip_revealed_nodes(
+    proof_nodes: alloy_trie::proof::DecodedProofNodes,
```

Comment:
let's maybe import it as-is and rename `DecodedProofNodes` in this file to just `ProofNodes`, as there's no decoding happens here anymore
```suggestion
    proof_nodes:DecodedProofNodes,
```

Reply:
ah yeah, will change

Reply:
ok, there is another `ProofNodes` type, going to change this to something else

Reply:
renamed to FilteredProofNodes